src/main/java/com/code4ro/legalconsultation/service/impl/pdf/PDFServiceImpl.java

```
@@ -36,4 +51,32 @@ public String readAsString(final byte[] file) {
             throw new LegalValidationException("document.parse.pdf.failed", HttpStatus.BAD_REQUEST);
         }
     }
+
+    @Override
+    public PdfHandle createPdf(DocumentConsolidated owner, String state, MultipartFile file) {
+        Integer hash = file.hashCode();
+        if (pdfHandleRepository.existsByHash(hash)) {
+            if (pdfHandleRepository.existsByHashAndState(hash, state)) {
+                return pdfHandleRepository.findByHashAndState(hash, state);
+            } else {
+                throw new RuntimeException(String.format("pdf document already exists with the state %s", state));
```

if this is an interesting message from the user's standpoint please throw a translatable exception like LegalValidationException. otherwise i thing just logging it should suffice.